### DISCUSSION TITLE
Reboot HSR in simulation without restarting simulator

### DISCUSSION THREAD

**Discussion by Kai-X-Org** [HumanSupportRobot/QandA#83] [Q&A]
Hi!

Is there a way to restart the HSR in the simulator without restarting the simulator itself? Thanks a lot!

**Comment by k-takeshita** [comment]
Using "/gazebo/reset_world" should work.
However, keep in mind that you may need to separately reset the self-position and other relevant information recognized by ROS nodes.

**Reply by Kai-X-Org** [reply]
Thanks a lot! But what I meant was is there a way to reboot the hsr's system in the simulation without restarting the entire simulator rather than just resetting its position. Apologies for not making it clear. Thanks a lot!

**Reply by k-takeshita** [reply]
Would Launching Gazebo, ROS nodes, and RViz separately be suitable for your purpose?

This is a simple example. Please set the args appropriately.

- `roslaunch hsrb_gazebo_bringup hsrb_empty_world.launch use_laser_odom:=false`
- `roslaunch hsrb_common_launch hsrb_common.launch sim:=true map:=/opt/ros/noetic/share/tmc_potential_maps/maps/white_space/map.yaml`
- `rviz -d /opt/ros/noetic/share/hsrb_common_launch/config/hsrb_display_full_hsrb.rviz`

**Reply by Kai-X-Org** [reply]
Thank you so much! Is there a way to perhaps stop and relaunch the nodes and the robot itself simply from Python code? Our projects involves rapid, continuous simulation that is meant to be able to be run overnight, so it would be great if no manual stopping and relaunching is necessary. Thank you so much! Apologies for the late reply.

**Reply by kosei1515** [reply]
In this module, we do not provide any Python scripts for your purpose. However, you could use `subprocess` or `rosnode` command to kill or relaunch nodes.
Here are the example of `subprocess`:
https://docs.python.org/3/library/subprocess.html
https://answers.ros.org/question/221077/killing-a-rosrun-process-using-python-script/

**Reply by Kai-X-Org** [reply]
Thank you very much! I will go and try this out!
